pom.xml

```diff
@@ -292,6 +292,19 @@
 				<groupId>org.apache.felix</groupId>
 				<artifactId>maven-bundle-plugin</artifactId>
 				<version>5.1.9</version>
+				<configuration>
+					<instructions>
+						<!-- 
+							Ensure JSON dependencies are not required at run-time in OSGi environments and also
+							allow usage of newer org.json versions.
+						-->
+						<Import-Package>
+							com.google.gson;resolution:=optional,
+							org.json;resolution:=optional;version="20240303.0",
```

Comment:
This indicates a fixed version. Does this mean whenever we update the version of org.json, we'd have to change here as well?

Reply:
My understanding is that this is an open range - `["20240303.0", Infinity)`. I'll dig up some references to it and post some screenshots of it working with different versions.

Reply:
According to https://docs.osgi.org/specification/osgi.core/7.0.0/framework.module.html#i3189032 

> If a version range is specified as a single version, it must be interpreted as the range [version,∞).

So this means just a minimal version. Right now org.json does not have a version newer that 20240303, but I will update the bundle locally to accept older versions for org.json so I can post a test with exact match and one with a newer version.

Reply:
Here's a more explicit test. I modified the Jedis bundle locally to reference a lower import range

```patch
diff --git a/pom.xml b/pom.xml
index f929411d..415273d4 100644
--- a/pom.xml
+++ b/pom.xml
@@ -300,7 +300,7 @@
 						-->
 						<Import-Package>
 							com.google.gson;resolution:=optional,
-							org.json;resolution:=optional;version="20240303.0",
+							org.json;resolution:=optional;version="20220303.0",
 							*
 						</Import-Package>
 					</instructions>

```

I then deployed it in my OSGi app and kept the same (new) 20240303 org.json bundle. In the screenshot you can see that the import is resolved at runtime with the new 20240303.0 version, even though the Import-Package declaration references 20220303.0 .

Effectively all versions newer than 20240303.0 are supported. If you want me I can lower the import range if you think that older versions can be used. Otherwise I'm happy to leave this as it is.

![image](https://github.com/user-attachments/assets/c43a8b45-98cd-463f-9b2b-99c97e7a5a51)

Reply:
Thank you @rombert for your investigation. It'll be helpful for us.